drivers/char/vhm/vhm_dev.c

```diff
@@ -395,6 +395,23 @@ static long vhm_dev_ioctl(struct file *filep,
 		break;
 	}
 
+	case IC_CLEAR_VM_IOREQ: {
+		/*
+		 * TODO: Query VM status with additional hypercall.
+		 * VM should be in paused status.
+		 *
+		 * In SMP SOS, we need flush the current pending ioreq dispatch
+		 * tasklet and finish it before clearing all ioreq of this VM.
+		 * With tasklet_kill, there still be a very rare race which
+		 * might lost one ioreq tasklet for other VMs. So arm one after
+		 * the clearing. It's harmless.
+		 */
+		tasklet_kill(&vhm_io_req_tasklet);
+		tasklet_schedule(&vhm_io_req_tasklet);
+		acrn_ioreq_clear_request(vm);
+		break;
+	}
+
```

Comment:
Is it necessary to add one ioctl to flush the pending io  request?
From the comment it seems that it should be in paused_state. If so, can we flush the pending request implicitly in reset ioctl?

Reply:
reset is too late to flush and clear the ioreqs.